diff --git a/src/arm/windows/init-by-logical-sys-info.c b/src/arm/windows/init-by-logical-sys-info.c
index 6ce6efe..d3c2279 100644
--- a/src/arm/windows/init-by-logical-sys-info.c
+++ b/src/arm/windows/init-by-logical-sys-info.c
@@ -644,6 +644,22 @@ static bool parse_relation_cache_info(
 	struct cpuinfo_cache* l2_base = l1d_base + numbers_of_caches[cpuinfo_cache_level_1d];
 	struct cpuinfo_cache* l3_base = l2_base + numbers_of_caches[cpuinfo_cache_level_2];
 
+#ifdef __MINGW32__
+	cpuinfo_log_debug(
+    		"info->Cache.GroupMask:%" PRIu32
+		","
+		"info->Cache.Level:%" PRIu32 ", info->Cache.Associativity:%" PRIu32
+		","
+		"info->Cache.LineSize:%" PRIu32
+		","
+		"info->Cache.CacheSize:%" PRIu32 ", info->Cache.Type:%" PRIu32 "",
+    		(unsigned int)info->Cache.GroupMask.Mask,
+    		info->Cache.Level,
+		info->Cache.Associativity,
+		info->Cache.LineSize,
+		info->Cache.CacheSize,
+		info->Cache.Type);
+#else
 	cpuinfo_log_debug(
 		"info->Cache.GroupCount:%" PRIu32 ", info->Cache.GroupMask:%" PRIu32
 		","
@@ -659,6 +675,7 @@ static bool parse_relation_cache_info(
 		info->Cache.LineSize,
 		info->Cache.CacheSize,
 		info->Cache.Type);
+#endif
 
 	struct cpuinfo_cache* current_cache = NULL;
 	switch (info->Cache.Level) {
@@ -701,6 +718,18 @@ static bool parse_relation_cache_info(
 		current_cache->flags = CPUINFO_CACHE_UNIFIED;
 	}
 
+#ifdef __MINGW32__
+	const uint32_t group_id = info->Cache.GroupMask.Group;
+	KAFFINITY group_processors_mask = info->Cache.GroupMask.Mask;
+	while (group_processors_mask != 0) {
+		const uint32_t processor_id_in_group = low_index_from_kaffinity(group_processors_mask);
+		const uint32_t processor_global_index = global_proc_index_per_group[group_id] + processor_id_in_group;
+
+		store_cache_info_per_processor(processors, processor_global_index, info, current_cache);
+
+		group_processors_mask &= ~(1 << processor_id_in_group);
+	}
+#else
 	for (uint32_t i = 0; i < info->Cache.GroupCount; i++) {
 		/* Zero GroupCount is valid, GroupMask still can store bits set.
 		 */
@@ -721,6 +750,7 @@ static bool parse_relation_cache_info(
 			group_processors_mask &= (group_processors_mask - 1);
 		}
 	}
+#endif
 	return true;
 }
 
diff --git a/include/cpuinfo.h b/include/cpuinfo.h
index 387611c..e1c0a4a 100644
--- a/include/cpuinfo.h
+++ b/include/cpuinfo.h
@@ -520,6 +520,8 @@ enum cpuinfo_uarch {
 	cpuinfo_uarch_falkor = 0x00400103,
 	/** Qualcomm Saphira. */
 	cpuinfo_uarch_saphira = 0x00400104,
+	/** Qualcomm Oryon. */
+	cpuinfo_uarch_oryon = 0x00400105,
 
 	/** Nvidia Denver. */
 	cpuinfo_uarch_denver = 0x00500100,
diff --git a/src/arm/windows/init.c b/src/arm/windows/init.c
index de2f6cc..fa6a9f3 100644
--- a/src/arm/windows/init.c
+++ b/src/arm/windows/init.c
@@ -65,6 +65,11 @@ static struct woa_chip_info woa_chips[woa_chip_name_last] = {
 			  2420000000,
 		  },
 		  {cpuinfo_vendor_arm, cpuinfo_uarch_cortex_x1, 3000000000}}},
+	/* Snapdragon(R) X Elite - X1E80100 - Qualcomm(R) Oryon(TM) CPU @ 3.4 GHz */
+	[woa_chip_name_qualcomm_x1e] = {
+		L"Snapdragon(R) X Elite - X1E80100 - Qualcomm(R) Oryon(TM) CPU",
+		woa_chip_name_qualcomm_x1e,
+		{{cpuinfo_vendor_arm, cpuinfo_uarch_oryon, 3400000000}}},
 	/* Ampere Altra */
 	[woa_chip_name_ampere_altra] = {
 		L"Ampere(R) Altra(R) Processor",
diff --git a/src/arm/windows/windows-arm-init.h b/src/arm/windows/windows-arm-init.h
index dc6e184..2f00e64 100644
--- a/src/arm/windows/windows-arm-init.h
+++ b/src/arm/windows/windows-arm-init.h
@@ -10,7 +10,8 @@ enum woa_chip_name {
 	woa_chip_name_microsoft_sq_3 = 2,
 	woa_chip_name_microsoft_sq_3_devkit = 3,
 	woa_chip_name_ampere_altra = 4,
-	woa_chip_name_unknown = 5,
+	woa_chip_name_qualcomm_x1e = 5,
+	woa_chip_name_unknown = 6,
 	woa_chip_name_last = woa_chip_name_unknown
 };
 
